### ISSUE TITLE
Gradio demo does not seem to work

### ISSUE THREAD

**Issue by mickelliu** [ttengwang/Caption-Anything#4]
I have put in my OpenAI API key and loaded one of the example images that has a corgi. I clicked on anywhere on the picture and nothing seems to show up in the chatbox.

**Comment by tensorboy** [comment]
The suggestions from GPT-4:

The AssertionError raised in the code is due to the fact that `self.image_embedding` is `None`. In the `base_segmenter.py` file, line 53 checks if `self.image_embedding` is not None, and raises an AssertionError if the condition fails:

```python
assert self.image_embedding is not None
```

This error occurs because the `self.image_embedding` attribute of `self.segmenter` is not set before calling the `inference` method. In the `app.py` file, you are setting the image of the segmenter but not setting the image embedding:

```python
model.segmenter.set_image(image_input)
```

In order to fix this issue, you need to ensure that the image_embedding attribute is properly set before calling the inference method.

First, check if the `set_image` implementation calculates and sets the `image_embedding` attribute in the segmenter. If not, you might need to adjust the implementation of the `set_image` method or create another method to set `self.image_embedding` for the segmenter.

Make sure to carefully review the entire codebase to see where and when `self.image_embedding` is supposed to be set, and ensure it is correctly calculated and assigned before the `inference` method is called.

**Comment by tensorboy** [comment]
python app.py --regular_box  --captioner blip2 --port 6086 --gradio_share --disable_reuse_features

works

**Comment by zjr2000** [comment]
Thank you for raising this issue.

The bug you encountered was due to the previous version of our code, which only extracted embeddings for user-uploaded images. The embeddings for example images were not extracted, causing an assertion error.

We have since resolved this issue. You can find the relevant details and fixes in this commit 28522a8. Please let us know if you have any further questions or concerns.
